core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -317,9 +318,25 @@ private[spark] class SparkSubmit extends Logging {
         args.ivySettingsPath)
 
       if (!StringUtils.isBlank(resolvedMavenCoordinates)) {
-        args.jars = mergeFileLists(args.jars, resolvedMavenCoordinates)
-        if (args.isPython || isInternal(args.primaryResource)) {
-          args.pyFiles = mergeFileLists(args.pyFiles, resolvedMavenCoordinates)
+        if (!isKubernetesCluster) {
```

Comment:
Tiny nit: don't negate the if? then the k8s logic is next to the check for a k8s cluster, and is a tiny bit clearer